craft_application/application.py

```diff
@@ -730,7 +730,18 @@ def _emit_error(
         craft_cli.emit.error(error)
 
     def _get_project_vars(self, yaml_data: dict[str, Any]) -> dict[str, str]:
-        """Return a dict with project variables to be expanded."""
+        """Return a dict with project variables to be expanded.
+
+        DEPRECATED: This method is deprecated and is not called by default.
```

Comment:
This was already unused.

We forgot to drop this in 5.0 when we added the project service.